### PR TITLE
[RW-2820][risk=no] Add error messaging for Concept Search

### PR DESCRIPTION
**Pull request by dolbeew**
Add error handling for api calls in concept search and display error messages for failing calls
[Error Messaging Doc](https://docs.google.com/presentation/d/1vVK0s9AxXKyhcgZJqWAWrqiitsG-RTwh16DOTp47zEY/edit#slide=id.g5b5ba1fbdc_0_16)

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] I have run and tested this change locally

### CODE REVIEW

File: ui/src/app/pages/data/concept/concept-homepage.tsx [hunk 1]
```diff
@@ -82,6 +82,17 @@ const styles = reactStyles({
...
+    background: colors.warning,
+    color: colors.white,
+    fontSize: '12px',
+    fontWeight: 500,
+    textAlign: 'left',
+    border: '1px solid #ebafa6',
+    borderRadius: '5px',
+    marginTop: '0.25rem',
+    padding: '8px',
+  },
```
Comment (freemabd) [inline ui/src/app/pages/data/concept/concept-homepage.tsx]: Should we make an generic error component that is reusable across all react components for error messaging?

Comment (gjuggler) [reply ui/src/app/pages/data/concept/concept-homepage.tsx]: +1 to this – this PR caught my eye, as I was running into some scenarios elsewhere in the Workbench UI where we don't have a clear / consistent way of displaying errors. A core component here would be great.

Comment (dolbeew) [reply ui/src/app/pages/data/concept/concept-homepage.tsx]: Agreed, we're already using this same template in cohort builder, cohort review and now concept search (maybe other places as well?).  I've filed a [ticket](https://precisionmedicineinitiative.atlassian.net/browse/RW-4406) for this and can follow-up on it.

**Comment by freemabd** [comment]
> Yeah, we could definitely use some clean-up, especially after moving surveys in with the other domains. I've filed a [ticket](https://precisionmedicineinitiative.atlassian.net/browse/RW-4405) to come back and clean-up the stuff we no longer need.

Is there an easy way to deploy locally and test these error messages?

**Comment by dolbeew** [comment]
Easiest way I've found is to change one of the parameter variable names to a string.

So change
`conceptsApi().getDomainInfo(namespace, id)`
to
`conceptsApi().getDomainInfo('namespace', id)`

Should cause a 404 error

**Pull request merged**